examples/rust/examples/maker/settings.rs

```diff
@@ -0,0 +1,115 @@
+use config::{Config, File};
+use ethers::prelude::{Address, LocalWallet, Wallet};
+use http::Uri;
+use rpassword::read_password;
+use serde::Deserialize;
+use std::fs::read_to_string;
+use std::io::{stdout, Write};
+use std::str::FromStr;
+use tonic::transport::{Certificate, ClientTlsConfig};
+
+#[derive(Deserialize)]
+struct InnerSettings {
+    node_endpoint: String,
+    quay_endpoint: String,
+    settlement_contract: String,
+    keystore: Option<String>,
+    private_key: Option<String>,
+    ca_root: Option<String>,
+    domain_name: Option<String>,
+    approve_tokens: bool,
+    quote_zeroes: bool,
+    magic_address: String,
+    usdc_address: String,
+    weth_address: String,
+    wbtc_address: String,
+    gmx_address: String,
+}
+
+pub struct Settings {
+    pub node_endpoint: String,
+    pub quay_endpoint: Uri,
+    pub settlement_contract: Address,
+    pub wallet: LocalWallet,
+    pub tls_config: ClientTlsConfig,
+    pub approve_tokens: bool,
+    pub quote_zeroes: bool,
+    pub magic_address: Address,
+    pub usdc_address: Address,
+    pub weth_address: Address,
+    pub wbtc_address: Address,
+    pub gmx_address: Address,
+}
+
+impl Settings {
+    pub fn load(file: &str) -> Self {
+        let settings = Config::builder()
+            .add_source(File::with_name(file))
+            .build()
+            .unwrap();
+
+        let inner: InnerSettings = settings.try_deserialize().unwrap();
+
+        // Local wallet, if the keystore was not given we need to request the private key instead.
+        let wallet = if let Some(keystore) = inner.keystore {
+            decrypt_keystore(&keystore)
+        } else {
+            fetch_private_key(inner.private_key)
+        };
+
+        // TLS Configuration, use a standard path unless provided with an alternate
+        let pem = if let Some(ca_root) = inner.ca_root {
+            read_to_string(ca_root).unwrap()
+        } else {
+            read_to_string("/etc/ssl/cert.pem").unwrap()
+        };
+
+        let domain_name = inner
+            .domain_name
+            .unwrap_or(String::from("exchange.valorem.xyz"));
```

Comment:
While we are here, lets also update the default

```suggestion
            .unwrap_or(String::from("trade.valorem.xyz"));
```